stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt

```diff
@@ -96,6 +96,7 @@ public class ChannelListViewModel(
         viewModelScope.launch {
             searchQuery.combine(queryConfig) { query, config -> query to config }
                 .collectLatest { (query, config) ->
+                    channelsState = channelsState.copy(searchQuery = query)
```

Comment:
Does it make sense to maybe push this towards the happy path - `is QueryChannelsController.ChannelsState.Result -> {}` in `observeChannels`?

Because this way we might have two state updates and UI prompts (compose should optimize for us but still).

Reply:
Nice catch, was also thinking about that. I will also cover not so happy path when the network in not available and we get `QueryChannelsController.ChannelsState.OfflineNoResults`